src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/CompilerServices/Utils.vb

```diff
@@ -301,9 +301,52 @@ Namespace Global.Microsoft.VisualBasic.CompilerServices
             End If
         End Function
 
+        ' s_MemberEquivalence will replace itself with one version or another
+        ' depending on what works at run time
+        Private s_MemberEquivalence As Func(Of MethodBase, MethodBase, Boolean) =
+            Function(m1, m2)
+                Try
+                    ' See if MetadataToken property is available.
+                    Dim MemberInfo As Type = GetType(MethodBase)
+                    Dim [property] As PropertyInfo = MemberInfo.GetProperty("MetadataToken", GetType(Integer), Array.Empty(Of Type)())
+
+                    If ([property] IsNot Nothing AndAlso [property].CanRead) Then
```

Comment:
This only detects availability, so reverting does not completely fix the problem that we can get `PlatformNotSupportedException` here.